src/Stratis.Bitcoin.Features.PoA.Tests/VotingManagerTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Stratis.Bitcoin.Features.PoA.Voting;
+using Stratis.Bitcoin.Tests.Common;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.PoA.Tests
+{
+    public class VotingManagerTests : PoATestsBase
+    {
+        private readonly VotingManager votingManager;
+
+        public VotingManagerTests()
+        {
+            string dir = TestBase.CreateTestDir(this);
+            var keyValueRepo = new KeyValueRepository(dir, new DBreezeSerializer(this.network));
```

Comment:
This are not integration tests I wonder if you should not mock the KeyValue repo?

Reply:
No, why? 
If I can use actual implementation then it's always better to do so. 
Things will be saved to temp folder anyway.

Reply:
Yeah but there is no added benefit in this case you are testing `KeyValueRepository` which  has its own tests, using a mock will skip db and made the subject being tested much faster.

Note this is for unit tests I refer to no integration tests.

Reply:
When actual implementation is used we are testing it indirectly which makes all tests more robust.